src/python/pants/bin/remote_pants_runner.py

```diff
@@ -132,14 +132,21 @@ def _connect_and_execute(
     ) -> ExitCode:
         global_options = self._bootstrap_options.for_global_scope()
 
+        # NB: If an OS string is not valid UTF-8, Python encodes the non-decodable bytes
+        #  as lone surrogates (see https://peps.python.org/pep-0383/).
+        #  However when we pass these to Rust, we will fail to decode as strict UTF-8.
+        #  So we perform a lossy re-encoding to prevent this.
+        def strict_utf8(s: str) -> str:
```

Comment:
Stepping away back, the ideal is to be lazy. This sort of conversion is not correct. You can imagine, if the env var pointed to a path for example, it now no longer does. For the scie side of these fixes, that's what I did. Only force the env vars you actually need (read and act on) to be utf-8. For those you don't care about, definitely don't muck since you know not what you may have broken.

Reply:
My memory was incorrect on how I fixed. I am lazy, but I warm, which doesn't help someone that really needs to deal with non-utf-8: https://github.com/a-scie/jump/pull/106#discussion_r1335039559

This stuff is hard to do right but at least the Rust OsString/OsStr apis keep you honest.

Reply:
Yeah, good point. My purpose here was not to support non-UTF-8 in env vars we consume, but to prevent unrelated env vars that we don't even consume from crashing us.  Possibly this should WARN though!

Actually supporting non-UTF-8 would be more work, as a lot of types will have to change in various places, so I'm unsure if it's worth it. I can take a stab at it on top of this though.

Reply:
Yeah, you saved a Pants crash but caused a tool to fail silentlyish by corrupting the env vars it uses. I agree that the window of env vars you'd corrupt that a tool actually uses may be small, but warning may be enough to allow the person in that narrow corner to see the issue and call home and complain that, for them, this is a bug that needs to be fixed.

Reply:
OK, fixed this up to WARN on such env vars, and ignore them. I think that's a better place to be than our current inability to run Pants at all even in the (likely?) case that the non-utf8 env var is not even consumed by Pants or any of its tools.

Reply:
Yeah, that's better. I don't agree with Huon that the user who hits the issue files a feature request - this is a straight up bug from day one, but this is better and feature / bug is a pedantic distinction at that point anyhow - it just places the pressure in the right spot I think.